mage_integrations/mage_integrations/destinations/gcs/gcs.py

```diff
@@ -0,0 +1,168 @@
+import os
+import logging
+from typing import Dict, List
+from datetime import datetime, timezone
+from io import BytesIO
+
+import pandas as pd
+from google.cloud import storage
+
+from mage_integrations.destinations.base import Destination
+from mage_integrations.destinations.utils import update_record_with_internal_columns
+
+
+class GoogleCloudStorage(Destination):
+    """
+    Export data to Google Cloud Storage.
+
+    Attributes:
+        bucket (str): The GCS bucket name to export data to.
+        file_type (str): The file type to export the data in, default is 'csv'.
+        object_key_path (str): The object key path in the bucket to export data to.
+        region (str): The GCS region to export data to, default is 'us-west1'.
+        client (google.cloud.storage.Client): The GCS client object used to interact with the API.
+    """    
+    @property
+    def bucket(self) -> str:
+        """
+        str: The GCS bucket name to export data to.
+        """
+        return self.config['bucket']
+
+    @property
+    def file_type(self) -> str:
+        """
+        str: The file type to export the data in, default is 'csv'.
+        """
+        return self.config.get('file_type', 'csv')
+
+    @property
+    def object_key_path(self) -> str:
+        """
+        str: The object key path in the bucket to export data to.
+        """
+        return self.config['object_key_path']
+
+    @property
+    def region(self) -> str:
```

Comment:
Is region needed for GCS?

Reply:
GCS location or region is optional, will remove it